grove/arbitrary_state/unitary_operator.py

```diff
@@ -0,0 +1,107 @@
+"""
+Module for creating a unitary operator for encoding any complex vector
+into the wavefunction of a quantum state. For example, the input vector
+:math:`[a, b, c, d]` would result in the state
+
+.. math::
+
+    a\\vert 00\\rangle + b\\vert01\\rangle
+    + c\\vert10\\rangle + d\\vert11\\rangle
+
+"""
+import numpy as np
+
+
+def unitary_operator(state_vector):
+    """
+    Uses QR factorization to create a unitary operator
+    that can encode an arbitrary normalized vector into
+    the wavefunction of a quantum state.
+
+    Assumes that the state of the input qubits is to be expressed as
+
+    .. math::
+
+        (1, 0, \\ldots, 0)^T
+
+    :param 1d array state_vector: Normalized vector whose length is at least
+                                  two and a power of two.
+    :return: Unitary operator that encodes state_vector
+    :rtype: 2d array
+    """
+
+    print np.linalg.norm(state_vector)
+    assert np.allclose([np.linalg.norm(state_vector)], [1]), \
```

Comment:
raise exceptions here and the next assert.  As this is not a "hidden" function a user could still pass in a violating state despite the API docstring specifying valid parameters.